scipy/stats/_qmc_cy.pyx

```diff
@@ -344,18 +413,62 @@ def _cy_van_der_corput(long n, long base, long start_index):
             b2r /= base
             quotient //= base
 
-    return sequence
 
 
-def _cy_van_der_corput_scrambled(long n, long base, long start_index,
-                                 long[:, ::1] permutations):
+cdef _cy_van_der_corput_scrambled(Py_ssize_t n,
+                                  long base,
+                                  long start_index,
+                                  long[:,::1] permutations,
+                                  unsigned int workers):
     sequence = np.zeros(n)
+    cdef double[::1] sequence_view = sequence
+
+    _cy_van_der_corput_scrambled_loops(n, base, start_index, permutations,
+                                       workers, sequence_view)
+
+    return sequence
+
+
+cdef _cy_van_der_corput_scrambled_loops(Py_ssize_t n,
+                                        long base,
+                                        long start_index,
+                                        long[:,::1] permutations,
+                                        unsigned int workers,
+                                        double[::1] sequence_view):
+    if workers <= 8:
```

Comment:
I am just seeing this. Why is that?

Reply:
Also when I try with 8, there is no speedup. But going to 10 I have x8 speedup. I just have 6 physical cores.

```python
%timeit van_der_corput(100_000, 2, start_index=0, scramble=True, workers=8)
53.8 ms ± 95.7 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
%timeit van_der_corput(100_000, 2, start_index=0, scramble=True, workers=10)
6.7 ms ± 66.5 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
53.8/6.7
Out[18]: 8.029850746268655
%timeit van_der_corput(100_000, 2, start_index=0, scramble=True, workers=12)
6.07 ms ± 290 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
%timeit van_der_corput(100_000, 2, start_index=0, scramble=True, workers=15)
8.31 ms ± 54.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
I think you spotted the mistake. Well done ahah ;)